projects/storefrontlib/src/cms-components/configurator/commons/attribute-header/attribute-header.component.ts

```diff
@@ -0,0 +1,36 @@
+import { Component, Input } from '@angular/core';
+import { Attribute, UiType } from '@spartacus/core';
+import { UIKeyGeneratorService } from '../service/ui-key-generator.service';
+@Component({
+  selector: 'cx-config-attribute-header',
+  templateUrl: './attribute-header.component.html',
+})
```

Comment:
For performance of change detection of components we try to use `changeDetection: ChangeDetectionStrategy.OnPush`, where possible.

It applies to all components.

Please note then in unit tests your `fixture.detectChanges` will run only once. Than this config of TestBed may be handy: `TestBed.overrideComponent` (See: https://medium.com/@juliapassynkova/how-to-test-onpush-components-c9b39871fe1e#7c25)

Reply:
Done